addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2739,6 +2739,194 @@ options.registry.ReplaceImage = options.Class.extend({
+            if (!ids || !widget.m2oModel) {
+                widget._value = "[]";
+                return;
+            }
+            ids = ids.split(',').map(id => parseInt(id));
+            const selectedRecords = await this._rpc({
+                model: widget.m2oModel,
+                method: 'read',
+                args: [ids, ['display_name']],
+            });
+            widget._value = JSON.stringify(selectedRecords);
+
+            const validIds = selectedRecords.map(record => record.id);
+            const rottenIds = ids.filter(id => !validIds.includes(id));
+            if (rottenIds) {
+                this.selectValue(false, widget._value, widget._methodsParams);
+            }
+        });
+    },
+
+    /**
+     * @see this.selectClass
+     */
+    selectValue(preivewMode, widgetValue, params) {
+        const recordData = JSON.parse(widgetValue);
+        const ids = [];
+        const codes = [];
+        for (const record of recordData) {
+            ids.push(record.id);
+            codes.push(record[params.callWith]);
+        }
+        if (ids.length && codes.length) {
+            this.$target[0].dataset[params.attributeName + 'Id'] = ids.join(',');
+            this.$target[0].dataset[params.attributeName] = codes.join(',');
+        } else {
+            delete this.$target[0].dataset[params.attributeName];
+            delete this.$target[0].dataset[params.attributeName + 'Id'];
+        }
+    },
+
+    /**
+     * @override
+     */
+    selectDataAttribute(preivewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (params.attributeName === 'visibility' && widgetValue === 'conditional') {
+            const collapse = this.$el.children('we-collapse')[0];
+            collapse.classList.add('active');
+            $(collapse).children('.o_we_collapse_toggler')[0].classList.add('active');
+        }
+    },
+
+    /**
+     * @override
+     */
+    onTargetShow: async function () {
+        this.$target[0].style = this.$target[0].previousStyle;
+    },
+
+    /**
+     * @override
+     */
+    onTargetHide: async function () {
+        this.$target[0].previousStyle = this.$target.style;
+        this.$target[0].style.display = 'none';
+    },
+
+    /**
+     * Creates the selectors to then apply the rules to the DOM
+     * @override
+    */
+    cleanForSave: async function () {
+        this.$target[0].style = this.$target[0].previousStyle;
+        delete this.$target[0].dataset['invisible'];
+        if (this.$target[0].dataset['visibility'] !== 'conditional') {
+            delete this.$target[0].dataset['visibilityId'];
+            return this._super(...arguments);
+        }
+        let visibilityId = '';
+        const onlyAttributes = [];
+        const hideAttributes = [];
+        for (const attr of this.$target[0].attributes) {
+            if (attr.name.startsWith('data-visibility-') && !attr.name.startsWith('data-visibility-rule') && !attr.name.endsWith('-id')) {
```

Comment:
Should not you loop through the dataset of the element instead of its attributes?

Reply:
Because we later on add those attributes as "data-attr-name" in a CSS rule, I thought it was easier to loop through the attributes to get their names in the correct format. I could loop on the dataset and replace the name, it would make the loop faster but would add the extra action of having to replace the string. I don't know which is faster.